controller/lib/hal/hal.h

```diff
@@ -52,13 +52,16 @@ class HalApi {
   // millis().
   void delay(uint32_t ms);
 
+	int analogRead(int pin);
+
   // TODO: Need at least one HAL_MOCK_METHOD.
 
 private:
 #ifdef TEST_MODE
   // Instance variables used when mocking HAL.
 
   uint32_t millis_ = 0;
+	int pin_values_[15] = {0};
```

Comment:
pre-commit let you indent this line with a tab?

Reply:
Oops, working on a new laptop and didn't set up my vimrc correctly yet... Fixed.

Reply:
But yeah, precommit *did* let me.